File: backend/src/main/java/com/back/domain/shelter/entity/Shelter.java
package com.back.domain.shelter.entity;


import com.back.domain.pet.entity.Pet;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Getter
@Table(name = "shelter")
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Shelter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "shelter_id")
    private Long id;

    @Column(name = "shelter_name", nullable = false)
    private String name;

    @Column(name = "shelter_address", nullable = false)
    private String address;

    @Column(name = "shelter_city")
    private String city;

    @Column(name = "shelter_state")
    private String state;

    @Column(name = "shelter_zip_code")
    private String zipCode;

    @Column(name = "shelter_phone")
    private String phone;

    @CreatedDate
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "shelter")
    private List<Pet> pets = new ArrayList<>();

    @Builder
    public Shelter(String name, String address, String city, String state, String zipCode, String phone) {
        this.name = name;
        this.address = address;
        this.city = city;
        this.state = state;
        this.zipCode = zipCode;
        this.phone = phone;
    }
}

File: backend/src/main/java/com/back/domain/shelter/entity/Shelter.kt
package com.back.domain.shelter.entity

import com.back.domain.pet.entity.Pet
import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@Table(name = "shelter")
@EntityListeners(AuditingEntityListener::class)
class Shelter(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "shelter_id")
    val id: Long = 0L,

    @Column(name = "shelter_name", nullable = false)
    var name: String,

    @Column(name = "shelter_address", nullable = false)
    var address: String,

    @Column(name = "shelter_city")
    var city: String?,

    @Column(name = "shelter_state")
    var state: String?,

    @Column(name = "shelter_zip_code")
    var zipCode: String?,

    @Column(name = "shelter_phone")
    var phone: String?,

    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdAt: LocalDateTime = LocalDateTime.now()
) {
    @OneToMany(mappedBy = "shelter", cascade = [CascadeType.ALL], orphanRemoval = true)
    val pets: MutableList<Pet> = mutableListOf()
}